src/tests/filter.test.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useDispatch, Provider } from 'react-redux';
+import { render, fireEvent, act } from '@testing-library/react';
+import configureStore from 'redux-mock-store';
+import { filter } from '../Redux/home/homeSlice';
+import Filter from '../components/Filter';
+
+const mockStore = configureStore([]);
+
+describe('Filter component', () => {
+  test('filter input changes and dispatches the correct action', () => {
+    const store = mockStore({});
+    const dispatch = jest.fn();
+    useDispatch.mockReturnValue(dispatch);
+
+    const { getByPlaceholderText } = render(
+      <Provider store={store}>
+        <Filter />
+      </Provider>,
+    );
+    const input = getByPlaceholderText('Enter search..');
+
+    act(() => {
+      fireEvent.change(input, { target: { value: 'test' } });
+    });
+
+    expect(dispatch).toHaveBeenCalledWith(filter('test'));
+  });
+});
```

Comment:
- [ ] Kindly correct the tests and add unit tests for the components. Unit testing for React Apps means testing an individual React Component. It avoids the bugs on your app.

> Create unit tests for pure functions in your Redux code.
> Create integration tests for your application using the React Testing Library.
> You may need to mock the access to the API, so that your tests don't send actual requests.
> You may need to mock the connection to the Redux Store.

![image](https://user-images.githubusercontent.com/58642949/217985751-6b0381cf-0203-442a-bf42-cc035183cdce.png)